### ISSUE TITLE
Wikilinks broken with inline code

### ISSUE THREAD

**Issue by HippieRocket** [dendronhq/dendron#2181]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [ ] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [ ] Lookup.  Dendron's Lookup Command
- [x] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [x] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

Wikilinks don't work with inline code. It can't be highlighted in workspace or rendered properly in preview. A link can be bold or italic but not in inline code.

## To Reproduce

1. There exists a note about `CHARACTER_LENGTH()` in MySQL. Try to link another note `reference.mysql.function.string-function.char_length` here.
2. Add a link `[[CHAR_LENGTH()|reference.mysql.function.string-function.char_length]]`. It will be highlighted in workspace.
3. Add backtick around the link, then it will break.

## Expected behavior

The wikilink is highlighted in workspace and rendered in preview as a clickable link with inline code format.

## Screenshots

![source](https://user-images.githubusercontent.com/56949546/149617571-bc860f3d-5502-417e-9d10-0226f7cdc1ee.png)

![preview](https://user-images.githubusercontent.com/56949546/149617586-69a3883f-71e6-44ba-b737-b4fc9b55e5e0.png)

## Desktop (please complete the following information)

- OS: Window 11, 22000.466

**Comment by HippieRocket** [comment]
Maybe everything in the backticks will be considered as plain code so can we make an exception for it?

**Comment by ScriptAutomate** [comment]
> Maybe everything in the backticks will be considered as plain code so can we make an exception for it?

Yeah, backticks makes it considered an in-line plain code text string, looking like `[[CHAR_LENGTH()|reference.mysql.function.string-function.char_length]]`. That functionality should stay as it is, otherwise you can run into funky situations where you want brackets within inline codeblocks.

Though, something could potentially be said for:

```
[[`CHAR_LENGTH()`|reference.mysql.function.string-function.char_length]]
```

Since then it would be self-contained, and still work as a wikilink, since it is in the left-hand label section? Right now, the backticks are seen as part of the string.

This would be a better alternative, since backticks are often used for examples of something meant to be rendered as plain text code. This is currently how gfm markdown links work, too.

- Example markdown (non-wikilink) link:

```
[`code_example()`](https://example.com)
```

- Example render: [`code_example()`](https://example.com)

**Comment by HippieRocket** [comment]
@ScriptAutomate It would be fancy if custom titles in wikilinks can be formatted, which hasn't been supported. It could be a solution for any need for formatting like `[[*Custom Title*|root.foo]]` and `[[**Custom Title**|root.foo]]

**Comment by SeriousBug** [comment]
I think this is definitely an enhancement we can implement. Let me rephrase the title.
